addons/pos_self_order/models/res_config_settings.py

```diff
@@ -52,12 +52,21 @@ def _onchange_pos_self_order_kiosk(self):
             self.is_kiosk_mode = True
             self.pos_module_pos_restaurant = False
             self.pos_self_ordering_pay_after = "each"
+            for pm in self.pos_payment_method_ids.filtered(lambda x: x.is_cash_count):
+                self.pos_payment_method_ids = [(3, pm.id)]
         else:
             self.is_kiosk_mode = False
 
             if not self.pos_module_pos_restaurant:
                 self.pos_self_ordering_service_mode = 'counter'
 
+    @api.onchange("pos_payment_method_ids")
+    def _onchange_pos_payment_method_ids(self):
+        if self.pos_self_ordering_mode == 'kiosk':
+            for pm in self.pos_payment_method_ids:
+                if pm.is_cash_count:
+                    raise ValidationError(_("You cannot add cash payment methods in kiosk mode."))
+
```

Comment:
i am not entirely convinced that this is needed, but, if you want to keep it, please find a way to not duplicate the code from `pos.config`